### DISCUSSION TITLE
Approach to Setting Redirects After Login

### DISCUSSION THREAD

**Discussion by pyrex41** [dillonkearns/elm-pages#453] [General]
If a user logs in at some point in their journey through an app / site, what are some suggested approaches to capture their last page prior to login, to redirect them back to where they were previously? My initial elm-oriented response is to capture this in the Model - Update - View architecture, in this case in Shared.model, but that seems like it may not be the most elm-pages-centric approach. Some other suggestions I've gotten are to store the information in a hidden form field, or to set a cookie that gets cleared after login. Would love to get thoughts from others on what may be the best approach here.

**Comment by dillonkearns** [comment]
Yes it's a great conversation, thanks for starting the thread!

Another approach is to use query params, so you could redirect to `/login?startpage=dashboard`, and then you could consume that query param `startpage` upon successful login to redirect back there.
